.azure-pipelines/third-party-test-suites.yml

```diff
@@ -0,0 +1,239 @@
+trigger: none
+pr: none
+
+schedules:
+- cron: "0 3 * * *"
+  displayName: Daily 3am (UTC) build
+  branches:
+    include:
+    - master
+  always: true
+
+# Global variables
+variables:
+  buildConfiguration: Release
+  dotnetCoreSdk5Version: 5.0.103
+  relativeTracerHome: /src/bin/windows-tracer-home
+  relativeArtifacts: /src/bin/artifacts
+  ddTracerHome: $(System.DefaultWorkingDirectory)/src/bin/dd-tracer-home
+  tracerHome: $(System.DefaultWorkingDirectory)/src/bin/windows-tracer-home
+  artifacts: $(System.DefaultWorkingDirectory)/src/bin/artifacts
+  ddApiKey: $(DD_API_KEY)
+  DD_DOTNET_TRACER_MSBUILD:
+  NugetPackageDirectory: $(System.DefaultWorkingDirectory)/packages
+  relativeNugetPackageDirectory: packages
+  dotnetToolTag: build-dotnet-tool
+  testServiceName: dd-trace-dotnet-third-party
+
+# Declare the datadog agent as a resource to be used as a pipeline service
+resources:
+  containers:
+  - container: dd_agent
+    image: datadog/agent
+    ports:
+    - 8126:8126
+    env:
+      DD_API_KEY: $(ddApiKey)
+      DD_INSIDE_CI: true
+
+stages:
+- stage: run_tests
+  dependsOn: []
+  jobs:
+  - job: Postgres
+    strategy:
+      matrix:
+        Ubuntu_Major5:
+          imageName: ubuntu-20.04
+          os: ubuntu
+          tagName: v5.0.7
+        Windows_Major5:
+          imageName: windows-2019
+          os: windows
+          tagName: v5.0.7
+          serverDirectory: npgsql/.build
+        Ubuntu_Major4:
+          imageName: ubuntu-20.04
+          os: ubuntu
+          tagName: v4.1.9
+        Windows_Major4:
+          imageName: windows-2019
+          os: windows
+          tagName: v4.1.9
+          serverDirectory: npgsql/.build
+
+    variables:
+      pg_major: 13
+      postgis_version: 3
+
+    pool:
+      vmImage: $(imageName)
+
+    # Enable the Datadog Agent service for this job
+    services:
+      dd_agent: dd_agent
+
+    steps:
+
+    - checkout: none
+
+    # Download the latest runner tool from the 'consolidated-pipeline' pipeline, from the master branch
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        source: 'specific'
+        project: 'dd-trace-dotnet'
+        pipeline: 54
+        runVersion: 'latestFromBranch'
+        # runBranch: 'refs/heads/master' # Restore once this is tested
+        runBranch: 'refs/heads/zach/test/third-party-npgsql'
```

Comment:
Is this a TODO for later?

Reply:
Yes, before merging to master I'll swap out the lines 😄 Until I merge this, I can't pull build artifacts from master, only this branch which has the required change.